enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -508,3 +553,16 @@ func (q *PriorityTaskScheduler) GetQueuedTaskReservations() []*scpb.EnqueueTaskR
 	defer q.mu.Unlock()
 	return q.q.GetAll()
 }
+
+// customResourceCount represents custom resource values in such a way that
+// floating point rounding errors are not accumulated as values are added and
+// subtracted over time.
+type customResourceCount int64
+
+func customResource(value float32) customResourceCount {
+	return customResourceCount(value * 1e6)
```

Comment:
It's been a while since I my computer architecture class, but IIRC multiplying floating point numbers doesn't affect their susceptibility to loss-of-precision because it's basically just changing the exponent, not adding more bits for the mantissa (does that sound correct?).

Can we get away with requiring that these be specified as integers for the time being?

Reply:
the ship might have sailed on requiring these to be integers because we wanted these to work the same as Bazel's [`local_extra_resources`](https://bazel.build/reference/command-line-reference#flag--local_extra_resources) which uses floats for the values.

re. this multiplication, some loss of precision is OK - we just need the conversion from float to int to be deterministic, so that we don't accumulate floating point errors when adding and subtracting the same resource values over time.